Updated TDZ definition (#16637)
mdn/content: files/en-us/web/javascript/reference/statements/let/index.md

### Temporal dead zone (TDZ)

> The _Temporal Dead Zone_ (TDZ) is the time between a `let` or `const` variable being hoisted until code execution reaches the point where it is initialized with a value.

Reviewer: Do people really understand it better when the word "hoist" is explicitly mentioned?

TBF, the fact that `var`s _are_ hoisted is the confusing part of JavaScript, and `let`/`const` exist to fix that and align JS with most other (non-lazy) languages. I'd prefer to not even mention "hoist" here—it's simply from the start of the scope to the declaration. Not necessarily "initialized with a value".

Reviewer 2: Probably not. Hoist tends to show up because this gets discussed as "by comparison with `var`". We have probably are passed the point where that needs to happen.

@ayushikolay I probably wasn't clear about the fix I wanted. However there are bigger problems than that with this section. I'll rewrite it and you and @Josh-Cena can perhaps then review.

While inside the TDZ, the variable has not been initialized with a value, and any attempt to access it will result in a {{jsxref("ReferenceError")}}.
The variable is initialized with a value when execution reaches the line of code where it was declared.

> If no initial value was specified with the variable declaration, it will be initialized with a value of `undefined`

Reviewer: ```suggestion
If no initial value was specified with the variable declaration, it will be initialized with a value of `undefined`.
```

Also mention that only `let` can be declared uninitialized?

Reviewer 2: > Also mention that only let can be declared uninitialized?

@Josh-Cena Really not sure it is necessary. I'll merge your fix for now, but I'm open to a proposal that is both succint and adds value :-)

Reviewer: Ah sorry, forgot we are already on the `let` page 😄 Never mind

### Temporal dead zone (TDZ)

> `Temporal Dead Zone`(TDZ) is the time since when the variable is hoisted and till it is initialised to some value, the time-lapse between the declaration and initialisation of a variable(s) is known as the **temporal dead zone**.

Reviewer 2: ```suggestion
The _Temporal Dead Zone_ (TDZ) is the time between a `let` or `const` variable being hoisted until till it is initialized to some value.
```

Author: Thanks for writing it in a convenient way.

> which will return a value of `undefined` if they are accessed before they
> are declared.

> 

Reviewer 2: Suggest you revert this and remove the following comment. The original was better.

Author: @hamishwillee This piece of code is actually the part of the Original one. Anyways, I have removed it. Please check.

Reviewer 2: Yes, I'm saying the text in the original one was better and should be put back.
